File: core/src/main/java/uk/lobsterdoodle/namepicker/storage/KeyValueStore.java
package uk.lobsterdoodle.namepicker.storage;

public interface KeyValueStore {
    boolean getBool(String key, boolean defaultValue);

    long getLong(String key, long defaultValue);

    String getString(String key, String defaultValue);

    boolean contains(String key);

    Edit edit();

    interface Edit {
        Edit put(String key, boolean value);

        Edit put(String key, long value);

        Edit put(String key, String value);

        Edit remove(String key);

        Edit clear();

        void commit();
    }
}

File: core/src/main/kotlin/uk/lobsterdoodle/namepicker/storage/KeyValueStore.kt
package uk.lobsterdoodle.namepicker.storage

interface KeyValueStore {
    fun getBool(key: String, defaultValue: Boolean): Boolean
    fun getLong(key: String, defaultValue: Long): Long
    fun getString(key: String, defaultValue: String): String
    operator fun contains(key: String): Boolean
    fun edit(): Edit

    interface Edit {
        fun put(key: String, value: Boolean): Edit
        fun put(key: String, value: Long): Edit
        fun put(key: String, value: String): Edit
        fun remove(key: String): Edit
        fun clear(): Edit
        fun commit()
    }
}
